sycl/test/check_device_code/hip/matrix/matrix-hip-bfloat16-float-test.cpp

```diff
@@ -0,0 +1,90 @@
+// REQUIRES: hip
+
+// RUN: %clangxx -fsycl-device-only -fsycl-targets=amd_gpu_gfx90a -DSYCL_EXT_ONEAPI_MATRIX_VERSION=4 -S -Xclang -emit-llvm %s -o -| FileCheck %s
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+using sycl::ext::oneapi::bfloat16;
+
+int main() {
+
+  buffer<bfloat16, 1> bufA(nullptr, range<1>(1));
+  buffer<bfloat16, 1> bufB(nullptr, range<1>(1));
+  buffer<float, 1> bufC(nullptr, range<1>(1));
+  buffer<float, 1> bufD(nullptr, range<1>(1));
+
+  queue q;
+
+  q.submit([&](handler &cgh) {
+    sycl::accessor<bfloat16, 1, sycl::access::mode::read_write,
+                   sycl::target::device>
+        accA(bufA, cgh);
+    sycl::accessor<bfloat16, 1, sycl::access::mode::read_write,
+                   sycl::target::device>
+        accB(bufB, cgh);
+    sycl::accessor<float, 1, sycl::access::mode::read_write,
+                   sycl::target::device>
+        accC(bufC, cgh);
+    sycl::accessor<float, 1, sycl::access::mode::read_write,
+                   sycl::target::device>
+        accD(bufD, cgh);
+
+    cgh.parallel_for<class row_row_m16n16k16>(
+        nd_range<2>({1, 64}, {1, 64}),
+        [=](nd_item<2> item) [[sycl::reqd_work_group_size(1, 1, 64)]] {
+          sycl::sub_group sg = item.get_sub_group();
+
+          joint_matrix<sub_group, float, use::accumulator, 16, 16> sub_c{};
+          joint_matrix<sub_group, bfloat16, use::a, 16, 16, layout::row_major>
+              sub_a{};
+          joint_matrix<sub_group, bfloat16, use::b, 16, 16, layout::row_major>
+              sub_b{};
+
+          joint_matrix_load(
```

Comment:
nit: I don't think you really need to have joint_matrix_load /store in these tests: The purpose is to test builtins and they don't call any. IMO would be better for simplifying the test.